src/EventStore.Core/Services/PersistentSubscription/PersistentSubscriptionService.cs

```diff
@@ -420,31 +424,55 @@ string user
 				onFail(startFromValidationError);
 				return;
 			}
+			
+			List<PersistentSubscription> subscribers;
+			var subscription = new PersistentSubscription(
+				new PersistentSubscriptionParams(
+					resolveLinkTos,
+					key,
+					eventSource,
+					groupName,
+					startFrom,
+					recordStatistics,
+					ToMessageTimeout(messageTimeoutMilliseconds),
+					maxRetryCount,
+					liveBufferSize,
+					bufferSize,
+					readBatchSize,
+					ToCheckPointAfterTimeout(checkPointAfterMilliseconds),
+					minCheckPointCount,
+					maxCheckPointCount,
+					maxSubscriberCount,
+					_consumerStrategyRegistry.GetInstance(namedConsumerStrategy, key),
+					_streamReader,
+					_checkpointReader,
+					new PersistentSubscriptionCheckpointWriter(key, _ioDispatcher),
+					new PersistentSubscriptionMessageParker(key, _ioDispatcher)));
 
-			RemoveSubscription(eventSource.ToString(), groupName);
-			RemoveSubscriptionConfig(user, eventSource.ToString(), groupName);
+			_subscriptionsById[key] = subscription;
 
-			CreateSubscriptionGroup(eventSource,
-				groupName,
-				resolveLinkTos,
-				startFrom,
-				recordStatistics,
-				maxRetryCount,
-				liveBufferSize,
-				bufferSize,
-				readBatchSize,
-				ToCheckPointAfterTimeout(checkPointAfterMilliseconds),
-				minCheckPointCount,
-				maxCheckPointCount,
-				maxSubscriberCount,
-				namedConsumerStrategy,
-				ToMessageTimeout(messageTimeoutMilliseconds)
-			);
+			if (_subscriptionTopics.TryGetValue(stream, out subscribers)) {
+				for (int i = 0; i < subscribers.Count; i++) {
+					var sub = subscribers[i];
+					if (sub.SubscriptionId == key) {
+						try {
+							sub.Shutdown();
+						} catch (Exception ex) {
+							Log.Error(ex, "Failed to shut down subscriber with id: {subscriptionId}", sub.SubscriptionId);
+						}
 
-			_config.Updated = DateTime.Now;
-			_config.UpdatedBy = user;
-			_config.Entries.Add(new PersistentSubscriptionEntry {
-				Stream = eventSource.ToString(), //'Stream' name kept for backward compatibility
+						subscribers[i] = subscription;
+						break;
+					}
+				}
+			} else {
```

Comment:
it seems that `subscription` is not being added to the `subscribers` list in the `else` part

actually, should we throw an exception in the `else` part? since the subscription should exist in `_subscriptionTopics` at this point.

Reply:
edit: this may no longer be relevant if we go for the change suggested here: https://github.com/EventStore/EventStore/pull/3533/files#r987722268